.vib/common-tests/ginkgo-utils/k8s.go

```diff
@@ -0,0 +1,260 @@
+package ginkgo_utils
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"io"
+	"regexp"
+	"strconv"
+	"time"
+
+	appsv1 "k8s.io/api/apps/v1"
+	batchv1 "k8s.io/api/batch/v1"
+	v1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	cv1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+// Cluster initialization functions
+
+// ClusterConfigOrDie builds the Kubernetes rest API configuration handler using a
+// given kubeconfig file
+func ClusterConfigOrDie(kubeconfig string) *rest.Config {
+	if kubeconfig == "" {
+		panic("kubeconfig must be supplied")
+	}
+
+	config, err := clientcmd.BuildConfigFromFlags("", kubeconfig)
+	if err != nil {
+		panic(err.Error())
+	}
+
+	return config
+}
+
+// StatefulSet functions
+
+// StsGetAvailableReplicas returns the available replicas in a
+// StatefulSet instance
+func StsGetAvailableReplicas(ss *appsv1.StatefulSet) int32 {
+	return ss.Status.AvailableReplicas
+}
+
+// StsScale scales a StatefulSet instance to the number of replicas
+func StsScale(ctx context.Context, c kubernetes.Interface, ss *appsv1.StatefulSet, count int32) (*appsv1.StatefulSet, error) {
+	name := ss.Name
+	ns := ss.Namespace
+
+	for i := 0; i < 3; i++ {
```

Comment:
```suggestion
	for i := 0; i < maxRetries; i++ {
```
Define `maxRetries` using `const` at the top.